src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -0,0 +1,173 @@
+                var paramsAfterCurrentParam = parameterNames.Except(paramsBeforeCurrentParam).ToList();
+                paramsAfterCurrentParam.Remove(parameterName);
+
+                // If the index is not `0`, there is a node before the current one for sure
+                // If the index is `0`, try to add the node after the `summary` node,
+                // only if any existing nodes are at the top level--this route will not
+                // be taken if the existing node is nested in another node
+                var summaryNode = GetParameterNodes(newDocComment, "summary").FirstOrDefault();
+                if (index != 0 || !paramNodes.Any() && summaryNode != null)
+                {
+                    // First, try to get the node before the param node so we know where to insert the new node
+                    SyntaxNode nodeBeforeNewParamNode =  GetLastParamNodeCorrespondingToParameterInList(paramNodes, paramsBeforeCurrentParam) ??
+                                    summaryNode;
+
+                    newDocComment = newDocComment.InsertNodesAfter(nodeBeforeNewParamNode, new[] { GetNewNode(parameterName, false) });
+                    continue;
+                }
+                
+                // At this point, the node has to go at the beginning of the comment
+                var nodeAfterNewParamNode = GetLastParamNodeCorrespondingToParameterInList(paramNodes.AsEnumerable().Reverse(), paramsAfterCurrentParam) ??
+                                                   newDocComment.ChildNodes().First();
+
+                // Adjust for doc comment marker before the node
+                if (nodeAfterNewParamNode != null)
+                {
+                    var paramNodeSiblings = nodeAfterNewParamNode.Parent.ChildNodes().ToList();
+                    var indexOfNode = paramNodeSiblings.IndexOf(nodeAfterNewParamNode);
+
+                    // set insert node to be the doc comment signifier of the closest param before the new node
+                    if (indexOfNode > 0 && paramNodeSiblings[indexOfNode - 1] is TXmlTextSyntax)
+                    {
+                        nodeAfterNewParamNode = paramNodeSiblings[indexOfNode - 1];
+                    }
+                }
+
+                var newNodeList = new[]
+                {
+                    // the last value will almost always be true, unless the node is embedded in another doc comment node
+                    GetNewNode(parameterName, nodeAfterNewParamNode == newDocComment.ChildNodes().First())
+                };
+                newDocComment = newDocComment.InsertNodesBefore(nodeAfterNewParamNode, newNodeList);
+            }
+
+            var newRoot = root.ReplaceNode(docCommentNode, newDocComment.WithAdditionalAnnotations(Formatter.Annotation));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private List<TXmlElementSyntax> GetParameterNodes(SyntaxNode docComment, string nodeName)
```

Comment:
You don't always get or return parameter nodes here.  (For example, when you pass "summary" in).  This should be more like GetElementNodes(docComment, string elementName)